src/Features/CSharp/Portable/Completion/CompletionProviders/OperatorsAndIndexer/ConversionCompletionProvider.cs

```diff
@@ -63,17 +63,26 @@ protected override ImmutableArray<CompletionItem> GetCompletionItemsForTypeSymbo
 
         private ImmutableArray<CompletionItem> GetUserDefinedConversionsOfType(SemanticModel semanticModel, ITypeSymbol container, bool containerIsNullable, int position)
         {
-            var allMembers = container.GetMembers();
-            var allExplicitConversions = from m in allMembers.OfType<IMethodSymbol>()
-                                         where
-                                             m.IsConversion() && // MethodKind.Conversion
-                                             m.Name == WellKnownMemberNames.ExplicitConversionName && // op_Explicit
-                                             m.Parameters.Length == 1 && // Malformed conversion operator may have more or less than one parameter
-                                             container.Equals(m.Parameters[0].Type) // Convert from container type to other type
-                                         let typeName = m.ReturnType.ToMinimalDisplayString(semanticModel, position)
-                                         // Lifted conversion https://docs.microsoft.com/hu-hu/dotnet/csharp/language-reference/language-specification/conversions#lifted-conversion-operators
-                                         select CreateSymbolCompletionItem(m, typeName, targetTypeIsNullable: containerIsNullable && m.ReturnType.IsStructType(), position);
-            return allExplicitConversions.ToImmutableArray();
+            using var _ = ArrayBuilder<CompletionItem>.GetInstance(out var builder);
+
+            var containerOrBaseType = (ITypeSymbol?)container;
+            while (containerOrBaseType is not null)
+            {
+                var allMembers = containerOrBaseType.GetMembers();
+                var allExplicitConversions = from m in allMembers.OfType<IMethodSymbol>()
+                                             where
+                                                 m.IsConversion() && // MethodKind.Conversion
+                                                 m.Name == WellKnownMemberNames.ExplicitConversionName && // op_Explicit
+                                                 m.Parameters.Length == 1 && // Malformed conversion operator may have more or less than one parameter
+                                                 containerOrBaseType.Equals(m.Parameters[0].Type) // Convert from container type to other type
+                                             let typeName = m.ReturnType.ToMinimalDisplayString(semanticModel, position)
+                                             // Lifted conversion https://docs.microsoft.com/hu-hu/dotnet/csharp/language-reference/language-specification/conversions#lifted-conversion-operators
```

Comment:
fwiw, i find this way of commenting just brutal to read.  breaking up expressions is just not done like this in roslyn.  it indicates to me that these shouldn't be exprssions as they're soo complex that they need annotating subexpressions to explain what's going on.